.github/workflows/e2e.yml

```diff
@@ -50,7 +50,7 @@ jobs:
       - name: Install devbox
         uses: jetify-com/devbox-install-action@v0.11.0
         with:
-          enable-cache: "true"
+          enable-cache: ${{ inputs.runs-on != 'self-hosted-ncn-dind' }}
```

Comment:
Why does this speed it up?

Reply:
Because restoring and saving nix cache from self-hosted runner took 4 minutes, nearly 2 minutes longer than just installing without cache... As an example with this PR took 2m37s to install devbox (https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/actions/runs/9792133640/job/27037342162?pr=786#step:4:1) and without this PR took 4m18s (https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/actions/runs/9791349347/job/27036701758?pr=784#step:4:1)